lib/aggregate/utils.js

```diff
@@ -8,6 +12,91 @@ function getErrorReply(errorMessage, code) {
   return {documents: [error]};
 }
 
+// Ranks a value's type for $min and $max operation. See comment for
+// compareValues for details.
+function typeRank(value) {
+  if (_.isNumber(value)) {
+    return 3;
+  } else if (_.isString(value)) {
+    return 4;
+  } else if (utils.isObjectId(value)) {
+    return 8;
+  } else if (_.isPlainObject(value)) {
+    return 5;
```

Comment:
The order was rearranged to make sure the `isObjectId` check is made before the object check, in case some operation accidentally turns an `ObjectID` instance into a plain object (e.g., by cloning). The way `ObjectId`s are structured, `utils.isObjectId` would still detect it as such. I've added a clarifying comment.